examples/step-74/step-laplace-FEIV.cc

```diff
@@ -0,0 +1,1159 @@
+        face_values[i].resize(n_q);
+        fe_iv.get_fe_face_values(i).get_function_values(solution,
+                                                        face_values[i]);
+      }
+    for (unsigned int q = 0; q < n_q; ++q)
+      jump[q] = face_values[0][q] - face_values[1][q];
+  }
+
+  template <int dim>
+  void SIPGLaplace<dim>::get_function_gradients(
+    const FEInterfaceValues<dim> &fe_iv,
+    const Vector<double> &        solution,
+    std::vector<Tensor<1, dim>>   face_gradients[2])
+  {
+    const unsigned n_q = fe_iv.n_quadrature_points;
+    for (unsigned i = 0; i < 2; ++i)
+      {
+        face_gradients[i].resize(n_q);
+        fe_iv.get_fe_face_values(i).get_function_gradients(solution,
+                                                           face_gradients[i]);
+      }
+  }
+
+  template <int dim>
+  void SIPGLaplace<dim>::get_function_gradient_jump(
+    const FEInterfaceValues<dim> &fe_iv,
+    const Vector<double> &        solution,
+    std::vector<Tensor<1, dim>> & gradient_jump)
+  {
+    const unsigned              n_q = fe_iv.n_quadrature_points;
+    std::vector<Tensor<1, dim>> face_gradients[2];
+    gradient_jump.resize(n_q);
+    for (unsigned i = 0; i < 2; ++i)
+      {
+        face_gradients[i].resize(n_q);
+        fe_iv.get_fe_face_values(i).get_function_gradients(solution,
+                                                           face_gradients[i]);
+      }
+    for (unsigned int q = 0; q < n_q; ++q)
+      gradient_jump[q] = face_gradients[0][q] - face_gradients[1][q];
+  }
+
+
+  template <int dim>
+  void SIPGLaplace<dim>::get_function_gradient_average(
+    const FEInterfaceValues<dim> &fe_iv,
+    const Vector<double> &        solution,
+    std::vector<Tensor<1, dim>> & gradient_average)
+  {
+    const unsigned              n_q = fe_iv.n_quadrature_points;
+    std::vector<Tensor<1, dim>> face_gradients[2];
+    gradient_average.resize(n_q);
+    for (unsigned i = 0; i < 2; ++i)
+      {
+        face_gradients[i].resize(n_q);
+        fe_iv.get_fe_face_values(i).get_function_gradients(solution,
+                                                           face_gradients[i]);
+      }
+    for (unsigned int q = 0; q < n_q; ++q)
+      gradient_average[q] = 0.5 * (face_gradients[0][q] + face_gradients[1][q]);
+  }
+
+
+  template <int dim>
+  void SIPGLaplace<dim>::setup_system()
+  {
+    dof_handler.distribute_dofs(fe);
+    DynamicSparsityPattern dsp(dof_handler.n_dofs());
+    DoFTools::make_flux_sparsity_pattern(dof_handler, dsp);
+    sparsity_pattern.copy_from(dsp);
+
+    system_matrix.reinit(sparsity_pattern);
+    solution.reinit(dof_handler.n_dofs());
+    system_rhs.reinit(dof_handler.n_dofs());
+    constraints.close();
+  }
+
+  template <int dim>
+  double SIPGLaplace<dim>::compute_penalty(const double h1, const double h2)
+  {
+    const double degree = std::max(1.0, static_cast<double>(fe.get_degree()));
+    return degree * (degree + 1.0) * 0.5 * (1.0 / h1 + 1.0 / h2);
+  }
```

Comment:
It might be a good idea to find better names than just h1 or h2, or at least add a comment that describes these arguments. Otherwise, reading this code from "top to bottom" becomes confusing here.

Reply:
> 
> 
> It might be a good idea to find better names than just h1 or h2, or at least add a comment that describes these arguments. Otherwise, reading this code from "top to bottom" becomes confusing here.

True! I have changed them to cell_extend_left and cell_extend_right. Do you think they are clear? Or do you have a better idea?